modules/optimization/src/userobjects/DensityUpdateTwoConstraints.C

```diff
@@ -100,7 +100,7 @@ DensityUpdateTwoConstraints::DensityUpdateTwoConstraints(const InputParameters &
                  "This parameter needs to be provided when including thermal sensitivity.");
   }
 
-  auto transient = dynamic_cast<Transient *>(_app.getExecutioner());
+  auto transient = dynamic_cast<TransientBase *>(_app.getExecutioner());
```

Comment:
Same here for the dynamic_cast

Reply:
The cast is used for error checking later on to make sure we have a transient executioner driving this.